cmd/query/app/apiv3/grpc_handler.go

```diff
@@ -32,20 +34,24 @@ func (h *Handler) GetTrace(request *api_v3.GetTraceRequest, stream api_v3.QueryS
 		return fmt.Errorf("malform trace ID: %w", err)
 	}
 
-	query := querysvc.GetTraceParameters{
-		GetTraceParameters: spanstore.GetTraceParameters{
-			TraceID:   traceID,
-			StartTime: request.GetStartTime(),
-			EndTime:   request.GetEndTime(),
+	query := querysvc.GetTraceParams{
+		TraceIDs: []tracestore.GetTraceParams{
+			{
+				TraceID: traceID.ToOTELTraceID(),
+				Start:   request.GetStartTime(),
+				End:     request.GetEndTime(),
+			},
 		},
 		RawTraces: request.GetRawTraces(),
 	}
-	trace, err := h.QueryService.GetTrace(stream.Context(), query)
+	getTracesIter := h.QueryService.GetTraces(stream.Context(), query)
+	traces, err := iter.FlattenWithErrors(getTracesIter)
```

Comment:
@yurishkuro is it fine to use flatten with errors here since we only want to return 1 trace?

Reply:
no, because the trace may still be large and returned in chunks (when raw=true).

Reply:
That's not the case today I believe (see https://github.com/jaegertracing/jaeger/blob/main/cmd/query/app/querysvc/v2/querysvc/service.go#L192-L194). Do we want to open a patch for this to not aggregate the trace if `RawTraces=false`?

Reply:
Yes, we discussed it there